src/internal/storage/fileset/buffer.go

```diff
@@ -8,118 +8,102 @@ import (
 )
 
 type Buffer struct {
-	additive map[string]*file
-	deletive map[string]*file
+	additive map[string]map[string]*file
+	deletive map[string]map[string]*file
 }
 
 type file struct {
-	path  string
-	parts map[string]*part
-}
-
-type part struct {
-	tag string
-	buf *bytes.Buffer
+	path string
+	tag  string
+	buf  *bytes.Buffer
 }
 
 func NewBuffer() *Buffer {
 	return &Buffer{
-		additive: make(map[string]*file),
-		deletive: make(map[string]*file),
+		additive: make(map[string]map[string]*file),
+		deletive: make(map[string]map[string]*file),
 	}
 }
 
-func (b *Buffer) Add(p, tag string) io.Writer {
-	p = Clean(p, false)
-	if _, ok := b.additive[p]; !ok {
-		b.additive[p] = &file{
-			path:  p,
-			parts: make(map[string]*part),
-		}
+func (b *Buffer) Add(path string, tag ...string) io.Writer {
```

Comment:
Same recommendation as above, but this is internal so I don't really care as much here.